src/Config.Binder/BinderOptions.cs

```diff
@@ -13,5 +15,10 @@ public class BinderOptions
         /// If true, the binder will attempt to set all non read-only properties.
         /// </summary>
         public bool BindNonPublicProperties { get; set; }
+
+        /// <summary>
+        /// When a required property isn't found the binder will throw an exception.
+        /// </summary>
+        public List<string> RequiredProperties { get; set; } = new List<string>();
```

Comment:
`ICollection<string>` is better here I think, we don't care about relative ordering